src/js/utils/permissionUtils.js

```diff
@@ -11,13 +11,18 @@ const canEditRequest = (currentUser, request, location) => {
     ActivityCode.APPROVE_REQUEST,
   );
 
-  // If the location supports request approval, only the requestor is able to edit
-  // and the request can't be approved / rejected when editing
   if (isApprovalRequired) {
-    return isUserRequestor &&
-      (isLocationDestination || isLocationOrigin) &&
-      request.statusCode !== RequisitionStatus.APPROVED &&
-      request.statusCode !== RequisitionStatus.REJECTED;
+    // If the location supports request approval, only the requestor is able to edit
+    // If the request is rejected, then it cannot be edited
+    if (request.statusCode !== RequisitionStatus.APPROVED) {
+      return isUserRequestor &&
+        (isLocationDestination || isLocationOrigin) &&
+        request.statusCode !== RequisitionStatus.REJECTED;
+    }
+    // If the request is approved, approver cannot edit request,
+    // but fulfilling user can edit it
```

Comment:
This makes no sense to me (I understand that it was requested to work that way), but this still makes no sense to me. I see the use case where I have one person doing approvals and fulfilling.